traitsui/editors/shell_editor.py

```diff
@@ -119,24 +119,24 @@ def update_editor(self):
                 for name, value in dic.items():
                     locals[name] = value
 
-    def update_any(self, event):
+    def update_any(self, object, name, old, new):
         """ Updates the editor when the object trait changes externally to the
             editor.
         """
         locals = self._shell.interpreter().locals
         if self._base_locals is None:
-            locals[event.name] = event.new
+            locals[name] = new
         else:
-            self.value[event.name] = event.new
+            self.value[name] = new
 
     def dispose(self):
         """ Disposes of the contents of an editor.
         """
-        self._shell.observe(
-            self.update_object, "command_executed", remove=True, dispatch="ui"
```

Comment:
hmm. why do we have `dispatch="ui"` with observe and not with `on_trait_change`? Looking at what the code used to be earlier.

Reply:
nvm. we're just reverting to what the code used to be - and `observe` requires the developer to specify `dispatch` if `dispatch` is set when we hook up the listener - but `on_trait_change` doesn't have this restriction/requirement.